src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc

```diff
@@ -83,6 +80,14 @@ ThreadedSocketConnection::~ThreadedSocketConnection() {
   }
 }
 
+void ThreadedSocketConnection::StopAndJoinThread() {
+  Disconnect();
+  thread_->join();
+  delete thread_->delegate();
+  threads::DeleteThread(thread_);
```

Comment:
@LuxoftAKutsan I have checked the DeleteThread method and no it doesn't deleate thread's delegate, but technically it is possible to move the "delete thread_->delegate()" method in it.